### ISSUE TITLE
Feature Request: Extract single file with unkown archive location

### ISSUE THREAD

**Issue by caigner** [borgbackup/borg#1829]
Assume there are several archives
```
archive1
archive2
archive3 <--- location of VeryOldFile.txt
archive4 <--- location of VeryOldFile.txt
archive5
archive6
.
.
.

```
Now let's assume I want to extract a file called "VeryOldFile.txt". I know that I created this file several months ago and I made a backup of it.

Recently I needed this file, but I realized that this file was deleted some months ago. I have backups of each week, but I don't know, when the file was deleted, so I don't know, when it was last backuped.

**Is there a way to find this file in the repository, other than looking into each archive manually?** 

If not I suggest the following: 

`borg extract /path/to/repo VeryOldFile.txt`

When no archive is specified start searching for the file beginning with youngest archive and go back in time until the file is found, then extract it.

A range option may be specified, like this:

`borg extract --search-within 6m /path/to/repo VeryOldFile.txt`

which would limit search to the backups of the last 6 months.

**Comment by enkore** [comment]
> Is there a way to find this file in the repository, other than looking into each archive manually?

```
borg mount <repository> <mountpoint>
find <mountpoint> -iname "VeryOldFile.txt"
```

Might take some time + memory.

**Comment by caigner** [comment]
Thanks! 

`borg mount <repository> <mountpoint>`

Works like a charm!

But now I wish for a more intuitive way to unmount than

`fusermount -u <mountpoint>`

Maybe something like

`borg umount <mountpoint>`

**Comment by infectormp** [comment]
`umount <mountpoint>`?

**Comment by enkore** [comment]
I think that needs sudo even if fusermount -u wouldn't normally.

**Comment by ThomasWaldmann** [comment]
FUSE "umount" is system-specific.

Some system let you do "umount" as user, linux wants you to do "fusermount -u" (as user).

**Comment by ThomasWaldmann** [comment]
In borg 1.1 we'll have a "versioned" FUSE mount, which will merge all (some) archive's files into one directory structure. So you'ld find all file versions in one place.

**Comment by ThomasWaldmann** [comment]
borg umount is implemented by #1856 (see also #1855).

**Comment by ThomasWaldmann** [comment]
closing this, mount or even versioned mount helps with this and the umount stuff is also done.
